zonemaster/zonemaster-engine
# Logger refactoring

## Purpose

This PR aims to provide an interface to manually set the module and test case name in the Logger.

This is to avoid unnecessary calls to `callers` and decouple the module / test case name from the file name / method name.

This PR also addresses the letter case of test cases and module to fit the updated specification.

## Context

https://github.com/zonemaster/zonemaster/pull/1200

## Changes

* In test modules, the local method `_emit_log` is used instead of `info`
* A somewhat global variable `$Zonemaster::Engine::Logger::TEST_CASE_NAME` is used to set the test case name
* Test case and module name have now the letter case defined in https://github.com/zonemaster/zonemaster/pull/1200 (including `Syestem` and `Unspecified`).
* Avoid building the caller trace and iterating through it for each log entry this may marginally improve performance.

## How to test this PR

Run a test using the CLI,

```
zonemaster-cli --level info --show-module --show-testcase afnic.fr
```
check that
* the module and test case are no longer full uppercase
* the log messages are correctly displayed and translated
